### PR TITLE
chore: Remove ancient NewRelic.Agent.Core subfolders.

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

Either drops the `NewRelic.Agent.Core` prefix or moves the files into more appropriate folders. Also updates some readme and script files that referenced a few of the moved files.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

**Review by chynesNR — APPROVED** [review]

File: src/Agent/NewRelic/Agent/Core/Core.csproj [hunk 1]
```diff
@@ -71,6 +71,13 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.0.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Remove="NewRelic.Agent.Core.Timing\**" />
+    <EmbeddedResource Remove="NewRelic.Agent.Core.Timing\**" />
+    <None Remove="NewRelic.Agent.Core.Timing\**" />
+    <Protobuf Remove="NewRelic.Agent.Core.Timing\**" />
+  </ItemGroup>
+
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Core/Core.csproj:80]: Is this still needed after the file renames?

File: src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs [hunk 2]
```diff
@@ -16,7 +15,7 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Net;
-using InternalMetricName = NewRelic.Agent.Core.Metric.MetricName;
+using InternalMetricName = NewRelic.Agent.Core.Metrics.MetricName;
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs:18]: Should we get a story added to the backlog to address this naming conflict that we are currently working around?

File: tests/Agent/UnitTests/Core.UnitTest/Aggregators/MetricAggregatorTests.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@
 using NewRelic.Agent.Core.DataTransport;
 using NewRelic.Agent.Core.Events;
 using NewRelic.Agent.Core.Fixtures;
-using NewRelic.Agent.Core.Metric;
+using NewRelic.Agent.Core.Metrics;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/Aggregators/MetricAggregatorTests.cs:13]: The duplicate using statements need to be cleaned up.

File: tests/Agent/UnitTests/Core.UnitTest/Aggregators/TransactionMetricStatsCollectionTests.cs [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2020 New Relic, Inc. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-using NewRelic.Agent.Core.Metric;
+using NewRelic.Agent.Core.Metrics;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/Aggregators/TransactionMetricStatsCollectionTests.cs:4]: The duplicate using statement needs to be cleaned up here too.

File: tests/Agent/UnitTests/Core.UnitTest/Transformers/MemorySampleTransformerTests.cs [hunk 5]
```diff
@@ -4,7 +4,7 @@
 using System.Collections.Generic;
 using NewRelic.Agent.Configuration;
 using NewRelic.Agent.Core.Aggregators;
-using NewRelic.Agent.Core.Metric;
+using NewRelic.Agent.Core.Metrics;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/Transformers/MemorySampleTransformerTests.cs:7]: This is another duplicate using statement that needs to be cleaned up.

File: tests/Agent/UnitTests/Core.UnitTest/Transformers/ThreadStatsSampleTransformerTests.cs [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 using System.Collections.Generic;
 using NewRelic.Agent.Core.Aggregators;
-using NewRelic.Agent.Core.Metric;
+using NewRelic.Agent.Core.Metrics;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/Transformers/ThreadStatsSampleTransformerTests.cs:6]: Another duplicate using statement.

File: tests/Agent/UnitTests/Core.UnitTest/Transformers/TransactionTransformer/TransactionTransformerTests.cs [hunk 7]
```diff
@@ -6,7 +6,7 @@
 using NewRelic.Agent.Core.CallStack;
 using NewRelic.Agent.Core.Errors;
 using NewRelic.Agent.Core.Metrics;
-using NewRelic.Agent.Core.Timing;
+using NewRelic.Agent.Core.Time;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/Transformers/TransactionTransformer/TransactionTransformerTests.cs:9]: Another duplicate statement.

File: tests/Agent/UnitTests/Core.UnitTest/WireModels/MetricBuilderTests.cs [hunk 8]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2020 New Relic, Inc. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-using NewRelic.Agent.Core.Metric;
+using NewRelic.Agent.Core.Metrics;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/WireModels/MetricBuilderTests.cs:4]: Another duplicate using statement.

File: tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs [hunk 9]
```diff
@@ -10,12 +10,11 @@
 using NewRelic.Agent.Core.Database;
 using NewRelic.Agent.Core.DistributedTracing;
 using NewRelic.Agent.Core.Errors;
-using NewRelic.Agent.Core.Metric;
 using NewRelic.Agent.Core.Metrics;
 using NewRelic.Agent.Core.Attributes;
 using NewRelic.Agent.Core.Segments;
 using NewRelic.Agent.Core.Segments.Tests;
-using NewRelic.Agent.Core.Timing;
+using NewRelic.Agent.Core.Time;
```
Comment (nrcventura) [inline tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs:17]: Duplicate using statement.

File: src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs (hunk 2 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs:18]: Yeah, I'll create one.

File: tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs (hunk 9 above)
Comment (jaffinito) [reply tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs:25]: Fixed

File: src/Agent/NewRelic/Agent/Core/Core.csproj (hunk 1 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Core.csproj]: Fixed

File: src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs (hunk 2 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs:18]: https://github.com/newrelic/newrelic-dotnet-agent/issues/1900

**Review by nrcventura — APPROVED** [review]

**Pull request merged**